C/c4ExpiryEnumerator.cc

```diff
@@ -103,9 +103,10 @@ bool c4exp_next(C4ExpiryEnumerator *e, C4Error *outError)
     return false;
 }
 
-C4Slice c4exp_getDocID(C4ExpiryEnumerator *e)
+C4SliceResult c4exp_getDocID(const C4ExpiryEnumerator *e)
```

Comment:
I think this bug happens because, in the implementation file, C4Slice is typedef'd to slice, and apparently there can be issues returning a non-vanilla C++ class/struct from a C function. Clang warns about this, but it didn't cause any problem at runtime so I just set a compiler flag to shut up the warning. Apparently with MSVC the ABI for C is different enough that it did screw things up. Your fix is probably for the best.